Source/DafnyCore/Backends/GoLang/GoBackend.cs

```diff
@@ -15,12 +16,30 @@ public class GoBackend : ExecutableBackend {
   public override string TargetName => "Go";
   public override bool IsStable => true;
   public override string TargetExtension => "go";
-  public override string TargetBaseDir(string dafnyProgramName) =>
-    $"{Path.GetFileNameWithoutExtension(dafnyProgramName)}-go/src";
+
+  public override string TargetBaseDir(string dafnyProgramName) {
+    var topLevelDir = $"{Path.GetFileNameWithoutExtension(dafnyProgramName)}-go";
+    if (GoModuleMode) {
+      return topLevelDir;
+    }
+
+    return $"{topLevelDir}/src";
+  }
 
   public override bool SupportsInMemoryCompilation => false;
   public override bool TextualTargetIsExecutable => false;
+
+  public bool GoModuleMode { get; set; } = true;
+  public string GoModuleName;
+
+  protected override IEnumerable<Option<string>> SupportedOptions => new List<Option<string>> { CommonOptionBag.BackendModuleName };
+
   protected override SinglePassCodeGenerator CreateCodeGenerator() {
+    Options.Options.OptionArguments.TryGetValue(CommonOptionBag.BackendModuleName, out var goModuleName);
```

Comment:
You can also write `var goModuleName = Options.Get(CommonOptionBag.BackendModuleName)`